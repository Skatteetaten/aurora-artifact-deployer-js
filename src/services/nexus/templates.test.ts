import {
  MavenSchema,
  LATEST_METADATA_XML,
  POM_XML,
  PROJECT_METADATA_XML,
} from './templates';

describe('Templates', (): void => {
  const config: MavenSchema = {
    artifactId: 'test',
    groupId: 'com.github.skatteetaten',
    lastUpdated: '20190826200800',
    packaging: 'zip',
    version: '1',
  };

  it('should create latest-metadata.xml', (): void => {
    expect(LATEST_METADATA_XML(config).toString()).toMatchSnapshot();
  });

  it('should create pom.xml', (): void => {
    expect(POM_XML(config).toString()).toMatchSnapshot();
  });

  it('should create project-metadata.xml', (): void => {
    expect(PROJECT_METADATA_XML(config).toString()).toMatchSnapshot();
  });
});
