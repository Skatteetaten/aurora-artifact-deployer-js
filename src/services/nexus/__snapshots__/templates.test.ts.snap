// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates should create latest-metadata.xml 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<metadata>
  <groupId>com.github.skatteetaten</groupId>
  <artifactId>test</artifactId>
  <version>1</version>
  <versioning>
    <lastUpdated>20190826200800</lastUpdated>
  </versioning>
</metadata>
"
`;

exports[`Templates should create pom.xml 1`] = `
"<project xmlns=\\"http://maven.apache.org/POM/4.0.0\\"
  xmlns:xsi=\\"http://www.w3.org/2001/XMLSchema-instance\\"
  xsi:schemaLocation=\\"http://maven.apache.org/POM/4.0.0
                      http://maven.apache.org/xsd/maven-4.0.0.xsd\\">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.github.skatteetaten</groupId>
  <artifactId>test</artifactId>
  <version>1</version>
  <packaging>zip</packaging>
</project>
"
`;

exports[`Templates should create project-metadata.xml 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<metadata>
  <groupId>com.github.skatteetaten</groupId>
  <artifactId>test</artifactId>
  <version>1</version>
  <versioning>
    <latest>1</latest>
    <versions>
      <version>1</version>
    </versions>
    <lastUpdated>20190826200800</lastUpdated>
  </versioning>
</metadata>
"
`;
